pkg/controller/controller.go

```diff
@@ -120,6 +150,8 @@ func (c *Controller) CreateVolume(ctx *c.Context, in *model.VolumeSpec, errchanV
 		AvailabilityZone: in.AvailabilityZone,
 		ProfileId:        profile.Id,
 		PoolId:           polInfo.Id,
+                SnapshotId:       in.SnapshotId,
+                SnapshotSize:     snapSize,
```

Comment:
Is it necessary to pass "SnapshotSize" as a parameter to CreateVolumeOpts and compare that to volume size when creating the db entry?   Can you do the comparison here and if size is not right, don't proceed to create a db entry?

Reply:
We allow creating a volume from a snapshot whose size is bigger than the snapshot. So it needs the size of the snapshot in the driver, for example, lvm.go. Or else it needs to query the size of the snapshot in lvm.go. How do you think?

Reply:
Got it.  Thanks!